client/src/features/action/needs/NeedFormGoods.tsx

```diff
@@ -10,11 +11,22 @@ import {
 } from '../../../assets/sharedComponents/Forms';
 import NeedOfferForm from '../NeedOfferForm';
 
-const categories = [
-  { value: 'figs', text: 'Figs' },
-  { value: 'peaches', text: 'Peaches' },
-  { value: 'pears', text: 'Pears' },
-];
+import type { Option } from '../../../types';
+
+const fetchCategories = async (): Promise<Option[]> => {
+  const res = await fetch('http://localhost:3001/api/categories?applies_to_assets=true');
+  const data = await res.json();
+
+  const categories = await data.map((category: any) => {
```

Comment:
don't need the `await` here
also let's make a `Category` type (can basically be a copy of the BE entity) to use here instead of `any`